src/Command/Generate/PluginMigrateDataParserCommand.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Console\Command\Generate\PluginMigrateDataParserCommand.
+ */
+
+namespace Drupal\Console\Command\Generate;
+
+use Drupal\Console\Utils\Validator;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Drupal\Console\Core\Command\ContainerAwareCommand;
+use Drupal\Console\Generator\PluginMigrateDataParserGenerator;
+use Drupal\Console\Command\Shared\ModuleTrait;
+use Drupal\Console\Extension\Manager;
+use Drupal\Console\Command\Shared\ConfirmationTrait;
+use Drupal\Console\Core\Utils\StringConverter;
+use Drupal\Console\Core\Utils\ChainQueue;
+
+class PluginMigrateDataParserCommand extends ContainerAwareCommand
+{
+    use ModuleTrait;
+    use ConfirmationTrait;
+
+    /**
+     * @var PluginMigrateProcessGenerator
+     */
+    protected $generator;
+
+    /**
+     * @var ChainQueue
+     */
+    protected $chainQueue;
+
+    /**
+     * @var Manager
+     */
+    protected $extensionManager;
+
+    /**
+     * @var StringConverter
+     */
+    protected $stringConverter;
+
+    /**
+     * @var Validator
+     */
+    protected $validator;
+
+    /**
+     * PluginMigrateDataParserGenerator constructor.
+     *
+     * @param PluginMigrateDataParserGenerator $generator
+     * @param ChainQueue                       $chainQueue
+     * @param Manager                          $extensionManager
+     * @param StringConverter                  $stringConverter
+     * @param Validator                        $validator
+     */
+    public function __construct(
+      PluginMigrateDataParserGenerator $generator,
+      ChainQueue $chainQueue,
+      Manager $extensionManager,
+      StringConverter $stringConverter,
+      Validator $validator
+    ) {
+        $this->generator = $generator;
+        $this->chainQueue = $chainQueue;
+        $this->extensionManager = $extensionManager;
+        $this->stringConverter = $stringConverter;
+        $this->validator = $validator;
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('generate:plugin:migrate:data_parser')
```

Comment:
@joshirohit100 Normally, underscore is not used in command name.
some examples generate:plugin:rulesaction, generate:plugin:rulesaction and etc.

Maybe it would be better to use something like this generate:plugin:migrate:data:parse. So, if we add also data fetcher we will have generate:plugin:migrate:data:fetcher. What do you think ?

Reply:
I am fine with both generate:plugin:migrate:dataparser or generate:plugin:migrate:data:parse
Lets see what other say

Reply:
I like `generate:plugin:migrate:dataparser` better

Reply:
Thanks! I have updated the PR